components/logins/src/login.rs

```diff
@@ -2,6 +2,253 @@
+//!
+//!   Applications should use the `touch()` method of the logins store to indicate when a password
+//!   has been used, and should ensure that they only count uses of the actual `password` field
+//!   (so for example, copying the `password` field to the clipboard should count as a "use", but
+//!   copying just the `username` field should not).
+//!
+//!   This number may not record uses that occurred on other devices, since some legacy
+//!   sync clients do not record this information. It may be zero for records obtained
+//!   via sync that have never been used locally.
+//!
+//!   When merging duplicate records, the largest non-zero value is taken.
+//!
+//!   This field is managed internally by the logins store by default and does not need to
+//!   be set explicitly, although any application-provided value will be stored.
+//!
+//!   If invalid data is received in this field (either from the application, or via sync)
+//!   then the logins store will attempt to coerce it into valid data by:
+//!   - replacing missing or negative values with 0
+//!
+//!   **XXX TODO:**
+//!   - test that we prevent this timestamp from moving backwards.
+//!   - test fixups of missing or negative values
+//!
+//! - `timeCreated`: An upper bound on the time of creation of this login, in integer milliseconds from the unix epoch.
+//!
+//!   This is an upper bound because some legacy sync clients do not record this information.
+//!
+//!   Note that this field is typically a timestamp taken from the local machine clock, so it
+//!   may be wildly inaccurate if the client does not have an accurate clock.
+//!
+//!   This field is managed internally by the logins store by default and does not need to
+//!   be set explicitly, although any application-provided value will be stored.
+//!
+//!   When merging duplicate records, the smallest non-zero value is taken.
+//!
+//!   If invalid data is received in this field (either from the application, or via sync)
+//!   then the logins store will attempt to coerce it into valid data by:
+//!   - replacing missing or negative values with the current time
+//!
+//!   **XXX TODO:**
+//!   - test that we prevent this timestamp from moving backwards.
+//!   - test fixups of missing or negative values
+//!
+//! - `timeLastUsed`: A lower bound on the time of last use of this login, in integer milliseconds from the unix epoch.
+//!
+//!   This is a lower bound because some legacy sync clients do not record this information;
+//!   in that case newer clients set `timeLastUsed` when they use the record for the first time.
+//!
+//!   Note that this field is typically a timestamp taken from the local machine clock, so it
+//!   may be wildly inaccurate if the client does not have an accurate clock.
+//!
+//!   This field is managed internally by the logins store by default and does not need to
+//!   be set explicitly, although any application-provided value will be stored.
```

Comment:
I don't think it's true that we allow application provided values, unless that changed recently. We could in the future, but the only use case I can think of is test code.